dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunctionLogExtensions.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.SemanticKernel.Functions;
+
+/// <summary>
+/// Extensions for logging <see cref="KernelFunction"/> invocations.
+/// This extension uses the <see cref="LoggerMessageAttribute"/> to
+/// generate logging code at compile time to achieve optimized code.
+/// </summary>
+internal static partial class KernelFunctionLogExtensions
+{
+    /// <summary>
+    /// Logs invocation of a <see cref="KernelFunction"/>.
+    /// </summary>
+    [LoggerMessage(
+        EventId = 0,
+        Level = LogLevel.Trace,
+        Message = "Function {FunctionName} invoking with arguments: {Arguments}.")]
+    public static partial void LogFunctionInvokingWithArguments(
+        this ILogger logger,
+        string functionName,
+        KernelArguments arguments);
+
+    /// <summary>
+    /// Logs cancellation of a <see cref="KernelFunction"/>.
+    /// </summary>
+    [LoggerMessage(
+        EventId = 1,
+        Level = LogLevel.Trace,
```

Comment:
I think that this message should have `Information` log level, since it doesn't contain sensitive information and should be useful to understand application flow, in this case - whether function was canceled or not. Same applies to method `LogFunctionComplete`.